### ISSUE TITLE
Deployment progress in xOpera CLI

### ISSUE THREAD

**Issue by mihaTrajbaric** [xlab-si/xopera-opera#191]
When orchestration, stack trace looks like this
```shell script
[Worker_0]   Deploying snow-security-rules_0
[Worker_0]     Executing create on snow-security-rules_0
[Worker_0]   Deployment of snow-security-rules_0 complete
[Worker_0]   Deploying snow-vm_0
[Worker_0]     Executing create on snow-vm_0
[Worker_0]   Deployment of snow-vm_0 complete
[Worker_0]   Deploying snow-docker-host_0
[Worker_0]     Executing create on snow-docker-host_0
[Worker_0]   Deployment of snow-docker-host_0 complete
[Worker_0]   Deploying snow-api_0
[Worker_0]     Executing create on snow-api_0
[Worker_0]   Deployment of snow-api_0 complete
```

It would be a useful feature to know deployment progress (which node of how many xOpera is processing at the moment), possibly like this:
```shell script
[Worker_0]  [Node 1/4]  Deploying snow-security-rules_0
[Worker_0]  [Node 1/4]     Executing create on snow-security-rules_0
[Worker_0]  [Node 1/4]   Deployment of snow-security-rules_0 complete
[Worker_0]  [Node 2/4]   Deploying snow-vm_0
[Worker_0]  [Node 2/4]     Executing create on snow-vm_0
[Worker_0]  [Node 2/4]   Deployment of snow-vm_0 complete
[Worker_0]  [Node 3/4]   Deploying snow-docker-host_0
[Worker_0]  [Node 3/4]     Executing create on snow-docker-host_0
[Worker_0]  [Node 3/4]   Deployment of snow-docker-host_0 complete
[Worker_0]  [Node 4/4]   Deploying snow-api_0
[Worker_0]  [Node 4/4]     Executing create on snow-api_0
[Worker_0]  [Node 4/4]   Deployment of snow-api_0 complete
```

**Comment by anzoman** [comment]
@mihaTrajbaric thanks, this would be a useful feature for the user to see how far the deployment goes.

**Comment by anzoman** [comment]
After some time, although it looks good at first, I think that this feature is not needed and could be just another distraction when running `opera deploy`. 

If we really need to show the deployment progress, I would recommend that we put it to the json/yaml output that is returned by our `opera info` command (we can add a new `progress` key that would show the percentage).

@cankarm, @dradx, @sstanovnik what do you think?

**Comment by sstanovnik** [comment]
I'd still keep this in a wishlist. In its current state, we don't have enough information to make good design decisions about how to implement progress. Some things we'd need to consider:

* progress of _what_ out of _what_
* will the progress work with more than one worker
* will we have progress reports from executors
* if so, will we have a high-pass filter for long-running tasks so we don't pollute the outputs
* is this even technically possible (I assume it is, but you never know)
* would we introduce a huge maintenance overhead because I don't think anything was designed to keep track of progress, and this is Python after all
* do we engage in ETC approximation

And lastly, like you suggested, we'd need to decide if and how to display progress via `info` for concurrent command execution (with e.g. xopera-api). Would we have a simple (quite useless) percentage or a more verbose data structure reporting stages.

In summary, I vote keep as wishlist. Not enough bang for our limited buck.
